build.sbt

```diff
@@ -46,23 +46,24 @@ def common: Seq[Setting[_]] = releaseSettings ++ bintraySettings ++ Seq(
       </developer>
     </developers>
   },
-  pomIncludeRepository := { _ => false },
-
+  pomIncludeRepository := { _ =>
+    false
+  },
   concurrentRestrictions in Global += Tags.limit(Tags.Test, 1),
-
   // Setting javac options in common allows IntelliJ IDEA to import them automatically
   javacOptions in compile ++= Seq(
-    "-encoding", "UTF-8",
-    "-source", "1.8",
-    "-target", "1.8",
+    "-encoding",
+    "UTF-8",
+    "-source",
+    "1.8",
+    "-target",
+    "1.8",
     "-parameters",
     "-Xlint:unchecked",
     "-Xlint:deprecation"
   ),
-
-  ScalariformKeys.preferences in Compile  := formattingPreferences,
-  ScalariformKeys.preferences in Test     := formattingPreferences,
-  ScalariformKeys.preferences in MultiJvm := formattingPreferences
+    scalafmtOnCompile in Compile := true,
+    scalafmtOnCompile in Test := true
```

Comment:
I have mixed feelings about formatting on compilation. 

From one side is a good thing because we avoid unformatted code to be pushed. Although, with proper checks (and we have) we can make the build to fail.

The problem I have with formatting on compile, is that I often let sbt runnning using `~`. From time to time I get an warning from IntelliJ that the file was changed on disk. That's quite annoying.

One advantage of scalafmt over scalariform, is that its plugins reads the same config file. That was not the case of scalariform. I have added a shortcut to IntelliJ to format the code using scalafmt configs and that simply works. You can even bind it to "on save". 

Obviously any quick edit in Sublime or Atom and push to an open PR will have the potential to break the build because of formatting, but I think that's fine.